lib/openstack_query/docs/developer_docs/ADDING_NEW_QUERIES.md

```diff
@@ -0,0 +1,236 @@
+# Adding a new Query
+The most common new feature for the Query Library would be adding the ability to query a new Openstack resource.
+You can do so by doing the following
+
+## 1. Create a PropEnum Class
+A new subclass of the `PropEnum` class is required to store valid property enums and mappings for your new openstack
+resource. To do this, you must:
+1. Identify all available properties for the new query.
+   - You can find this information by looking in the openstacksdk api documentation [here](https://docs.openstack.org/openstacksdk/latest/user/index.html#api-documentation).
+2. Create a new file under `/enums/query/props/` called `<resource>_properties.py`, replace `<resource>` with the name of the openstack resource you want to query
+3. Create the `<resource>Properties` class definition
+   - class must inherit from `PropEnum` abstract base class and implement abstract methods.
+
+```python
+class ResourceProperties(PropEnum):
+
+    # add enums for each property you found here like so
+    PROP_1 = auto() # replace `PROP_1` with property name
+    ...
+
+    @staticmethod
+    def get_prop_mapping(prop):
+        ...
+        # add prop mappings here.
+        mapping = {
+           ResourceProperties.PROP_1: lambda a: a["prop_1"]
+        }
+        try:
+           return mapping[prop]
+        except KeyError as exp:
+            raise QueryPropertyMappingError(
+                f"Error: failed to get property mapping, property {prop.name} is not supported in FlavorProperties"
+            ) from exp
```

Comment:
Can we move this to the parent class at some point, I dislike this exact code (minus the name) is now documented to be copy pasted into each one

Reply:
If I recall - you had a draft PR to update this at some point? 
If you're not working on this I''ll get on this soon